joomla/joomla-cms
# [plg_system_logout] Load language files only when needed

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/11714 (part 3).
### Summary of Changes

Only load the plg_system_logout language file when they are needed.

For more info see https://github.com/joomla/joomla-cms/issues/11714
### Testing Instructions
#### Reproduce
- Use latest staging
- Enable plugin system logout (if not yet)
- Enable debug and debug lang in global config
- Enable debug system plugin with all options activated
- Go to frontend and check the debug console
- Notice `JROOT/administrator/language/en-GB/en-GB.plg_system_logout.ini` is loaded no matter what page you are (even in backend pages they are loaded).

![image](https://cloud.githubusercontent.com/assets/9630530/17866131/468f6d38-689d-11e6-929a-c08337c51f7f.png)
#### Test patch
- Apply patch
- Repeat the steps above and check the language file is NOT loaded (except when needed). 
- Check plugin work as before if language files are required to load (ex: add `JError::raiseError(403, 'test');` after https://github.com/joomla/joomla-cms/pull/11736/files#diff-0bd8e36ff5e9b432eb9c55d94266d786R43 and them logout).
### Documentation Changes Required

None.
### Notes

This also solves a PHP warning when handling error on logout
`Warning: Parameter 1 to PlgSystemLogout::handleError() expected to be a reference, value given in /path/to/joomla-staging/libraries/legacy/error/error.php on line 781`

@mbabker please check this one also. Maybe this could be done without JError...
